usaspending_api/download/filestreaming/download_generation.py

```diff
@@ -450,6 +432,57 @@ def apply_annotations_to_sql(raw_query, aliases):
     return sql.replace(NAMING_CONFLICT_DISCRIMINATOR, "")
 
 
+def _select_columns(sql):
+    in_quotes = False
+    parens_depth = 0
+    last_processed_index = 0
+    retval = []
+
+    for index, char in enumerate(sql):
+        if char == '"':
+            in_quotes = not in_quotes
+        if in_quotes:
+            continue
+        if char == "(":
+            parens_depth = parens_depth + 1
+        if char == ")":
+            parens_depth = parens_depth - 1
+
+        if parens_depth == 0:
+            # Ignore the SELECT statement
+            if sql[index : index + 6] == "SELECT":
+                last_processed_index = last_processed_index + 6
+            # If there is a FROM at the bottom level, we have all the values we need and can return
+            if sql[index : index + 4] == "FROM":
+                retval.append(sql[last_processed_index:index].strip())
+                return retval
+            # If there is a comma on the bottom level, add another select value and start parsing a new one
+            if char == ",":
+                retval.append(sql[last_processed_index:index].strip())
+                last_processed_index = index + 1  # skips the comma by design
+
+    return retval  # this will almost certainly error out later.
```

Comment:
Would raising an Exception be better? Or are there legitimate workflows which return here?